### ISSUE TITLE
Don't leak thread stack on exit

### ISSUE THREAD

**Issue by tbu-** [japaric/steed#131]
Current implementation does not bother with freeing the child thread's resources.

**Comment by japaric** [comment]
Isn't this equivalent to #134? You need to implement `Drop` for `Thread` to not leak the stack, right?

**Comment by tbu-** [comment]
No, this is orthogonal. `Drop` on the thread handle should detach the thread (whatever that means) and destructing thread-local data must happen on thread exit.
